### PR TITLE
Santa costumes! 2

### PR DESCRIPTION
**Pull request by Termineitor244**
#### Summary
Content "Adds santa costumes to the game"

#### Purpose of change
It's already December! We need to have some santas delivering gifts to all the zombies! (And some zombies delivering gifts to the players).
Second try at #53097, since the original PR got messed up...

#### Describe the solution
Adds the clasic santa costume, as well as some other variants for more flavor.

This PR uses the event-based item spawns implemented in #53100 to bring extra Christmas spawns to the game, if you activate the toggle on the menu and wait until Christmas then you will see a lot of zombies dropping dead with all variants of Santa clothing on them, maybe just a hat, maybe a hat and the white beard, or maybe the full set of santa clothing!

#### Describe alternatives you've considered
To have a sad christmas in the middle of the cataclysm.

#### Testing
Started the game, killed zombies, searched houses... I did not found anything Santa related (But it should be possible to find with enough tries)...
Changed the date to 24 of December, tried the same, and suddenly the zombies are wearing Santa hats and some even a full set of Santa costume clothes!

#### Additional context
I would like to have more possible sets/clothing for Christmas special drops, but I'm not sure if I will have the time to implement more, the item groups were made so it would be easy to add spawns for random Christmas clothing or more sets as well, so it should be easier if anyone wants to give it a try.

Special thanks to @dseguin for its work on #53100! 
<details> 
<summary>Inspirations for the clothing (Images) </summary>

Santa traditional costume:
Santa suit long

Santa beard:
https://www.amazon.com/Costume-Christmas-Accessories-Adults-Disguise/dp/B07YWMSR2K/ref=sr_1_11?keywords=santa+beard&qid=1638173459&sr=8-11

Santa dress:
Santa dress normal

Santa luxury dress:
santa dress long luxury
Santa dress long luxury 2

Santa short dress:
Santa dress short

Santa short jacket:
Santa suit short

Santa short:
Santa shorts

</details>

### CODE REVIEW

File: data/json/monsterdrops/zombie_default.json [hunk 1]
```diff
@@ -26,7 +27,13 @@
                     "collection": [
                       { "group": "pants_male", "prob": 60, "damage": [ 1, 4 ] },
                       { "group": "shirts_unisex", "prob": 60, "damage": [ 1, 4 ] },
-                      { "item": "leather_belt", "prob": 30, "damage": [ 1, 4 ] }
+                      {
+                        "distribution": [
+                          { "item": "leather_belt", "prob": 10, "damage": [ 1, 4 ] },
+                          { "group": "christmas_belts", "prob": 90, "damage": [ 1, 4 ], "event": "christmas" }
```
Comment (Maleclypse) [inline data/json/monsterdrops/zombie_default.json:33]: Even during christmas 90% of the zombies shouldn't be wearing christmas items.   I would think.

File: data/json/monsterdrops/zombie_default.json [hunk 2]
```diff
@@ -76,6 +83,7 @@
                           { "item": "gown", "prob": 10, "damage": [ 1, 4 ] },
                           { "item": "dress", "prob": 70, "damage": [ 1, 4 ] },
                           { "item": "sundress", "prob": 45, "damage": [ 1, 4 ] },
+                          { "group": "christmas_dresses", "prob": 150, "damage": [ 1, 4 ], "event": "christmas" },
```
Comment (Maleclypse) [inline data/json/monsterdrops/zombie_default.json:86]: Same issue here.   Is there a debug test to pretend it's a current event so you can test drops or do we have to test drops once a year.

File: data/json/monsterdrops/zombie_default.json [hunk 3]
```diff
@@ -250,7 +258,12 @@
     "//": "contains cloth (always), a wallet (30% chance), and a random item (50% chance)",
     "id": "default_zombie_death_drops",
     "entries": [
-      { "group": "default_zombie_clothes" },
+      {
+        "distribution": [
+          { "group": "default_zombie_clothes", "prob": 90 },
+          { "group": "default_zombie_clothes_christmas", "event": "christmas", "prob": 10 }
```
Comment (Maleclypse) [inline data/json/monsterdrops/zombie_default.json:264]: This seems much more reasonable :)

**Comment by Fris0uman** [comment]
Kinda out of scope/off topic comment: It would be cool if those holyday themed things could spawn according to the start of the game: 
> winter start: plenty of santa and Xmass stuff around
> spring start: Xmass stuff are in boxes in basements

File: data/json/monsterdrops/zombie_default.json (hunk 2 above)
Comment (Termineitor244) [reply data/json/monsterdrops/zombie_default.json]: Well, I just had to test it while changing the date of my computer haha.

**Comment by Termineitor244** [comment]
@maleclypse these are the new results of a test item group spawn for the zombie clothing drops on Christmas (while searching for "santa" in the name), they include the 10% probability of a full set:

imagen

I reduced all the spawns, tell me if it these numbers are better (And sorry for the late reply, I didn't have much time lately).

 

> Kinda out of scope/off topic comment: It would be cool if those holyday themed things could spawn according to the start of the game:
> 
>     * winter start: plenty of santa and Xmass stuff around
> 
>     * spring start: Xmass stuff are in boxes in basements

Good ideas! I would have liked to implement the boxes already, but a quick search in the item groups folder didn't find the item group that boxes in basements use, maybe I should search in the mapgen folder.

File: data/json/items/armor/coats.json [hunk 4]
```diff
@@ -1565,6 +1565,65 @@
     "flags": [ "VARSIZE", "OUTER" ],
     "armor": [ { "encumbrance": 2, "coverage": 90, "covers": [ "leg_l", "leg_r", "torso" ] } ]
   },
+  {
+    "id": "santa_jacket",
+    "type": "ARMOR",
+    "name": { "str": "santa jacket" },
+    "description": "A white faux fur trimmed red jacket commonly used by bearded men on Christmas.  Warm and with a couple of small pockets.",
```
Comment (I-am-Erk) [inline data/json/items/armor/coats.json:1572]:
```suggestion
    "description": "A red jacket, trimmed with white faux fur, commonly used by bearded men on Christmas.  Warm and has a couple of small pockets.",
```

File: data/json/items/armor/coats.json [hunk 5]
```diff
@@ -1565,6 +1565,65 @@
...
+    "warmth": 40,
+    "material_thickness": 2,
+    "environmental_protection": 2,
+    "flags": [ "VARSIZE", "POCKETS", "OUTER" ]
+  },
+  {
+    "id": "santa_jacket_short",
+    "type": "ARMOR",
+    "name": { "str": "short santa jacket" },
+    "description": "A short red jacket with white faux fur as trim, similar to a Santa jacket but sleeveless.  This version is for those who seek a sexier Santa on Christmas.",
```
Comment (I-am-Erk) [inline data/json/items/armor/coats.json:1611]:
```suggestion
    "description": "A short red jacket with white faux fur as trim, similar to a Santa jacket but sleeveless, for a more revealing look on Christmas.",
```

File: data/json/items/armor/gloves.json [hunk 6]
```diff
@@ -713,6 +713,25 @@
     "flags": [ "VARSIZE", "WATER_FRIENDLY", "SKINTIGHT", "STURDY" ],
     "armor": [ { "encumbrance": 6, "coverage": 100, "covers": [ "hand_l", "hand_r" ] } ]
   },
+  {
+    "id": "santa_gloves",
+    "type": "ARMOR",
+    "name": { "str": "pair of santa gloves", "str_pl": "pairs of santa gloves" },
+    "description": "A pair of white faux fur trimmed red gloves.  Warm and for use on Christmas.",
```
Comment (I-am-Erk) [inline data/json/items/armor/gloves.json:720]:
```suggestion
    "description": "A pair of  red gloves with white faux fur trim.  Keeps your hands warm and your holiday spirit bright.",
```

File: data/json/items/armor/hats.json [hunk 7]
```diff
@@ -509,6 +509,24 @@
     "flags": [ "SUN_GLASSES" ],
     "armor": [ { "encumbrance": 10, "coverage": 60, "covers": [ "head" ] } ]
   },
+  {
+    "id": "santa_hat",
+    "type": "ARMOR",
+    "name": { "str": "santa hat" },
+    "description": "A white faux fur trimmed red hat commonly used by bearded men on Christmas.",
```
Comment (I-am-Erk) [inline data/json/items/armor/hats.json:516]:
```suggestion
    "description": "A red hat with white faux fur trim, commonly used by bearded men on Christmas.",
```

File: data/json/items/armor/legs_clothes.json [hunk 8]
```diff
@@ -917,6 +917,83 @@
...
+        "specifically_covers": [ "leg_hip_l", "leg_hip_r" ]
+      },
+      { "coverage": 40, "covers": [ "leg_l", "leg_r" ], "specifically_covers": [ "leg_upper_l", "leg_upper_r" ] }
+    ]
+  },
+  {
+    "id": "santa_pants",
+    "type": "ARMOR",
+    "name": { "str_sp": "santa pants" },
+    "description": "A pair of white faux fur trimmed red pants commonly used by bearded men on Christmas.  Somewhat warm and with two deep pockets.",
```
Comment (I-am-Erk) [inline data/json/items/armor/legs_clothes.json:951]:
```suggestion
    "description": "A pair of red pants with white faux fur trim, commonly used by bearded men on Christmas.  Somewhat warm and has two deep pockets.",
```

File: data/json/items/armor/torso_clothes.json [hunk 9]
```diff
@@ -574,6 +574,89 @@
     "flags": [ "VARSIZE" ],
     "armor": [ { "coverage": 90, "covers": [ "torso" ] } ]
   },
+  {
+    "id": "santa_dress",
+    "repairs_like": "jeans",
+    "type": "ARMOR",
+    "name": { "str": "santa dress", "str_pl": "santa dresses" },
+    "description": "A red sleeved dress with white faux fur as trim.  For when you want to be Mrs.  Claus on Christmas.",
```
Comment (I-am-Erk) [inline data/json/items/armor/torso_clothes.json:582]:
```suggestion
    "description": "A red sleeved dress with white faux fur as trim.  For when you want to be Mrs. Claus on Christmas, or perhaps Santa going clubbing.",
```

File: data/json/items/armor/torso_clothes.json [hunk 10]
```diff
@@ -574,6 +574,89 @@
...
+      { "covers": [ "arm_l", "arm_r" ], "coverage": 85, "encumbrance": 6 },
+      { "covers": [ "leg_l", "leg_r" ], "coverage": 100, "encumbrance": 8 }
+    ]
+  },
+  {
+    "id": "santa_dress_short",
+    "repairs_like": "jeans",
+    "type": "ARMOR",
+    "name": { "str": "short santa dress", "str_pl": "short santa dresses" },
+    "description": "A short red dress with white faux fur as trim.  For when you want to be a sexy Mrs.  Claus on Christmas.",
```
Comment (I-am-Erk) [inline data/json/items/armor/torso_clothes.json:636]:
```suggestion
    "description": "A short red dress with white faux fur as trim.  Looks like something Mrs. Claus might wear on Christmas night.",
```

**Review by I-am-Erk — CHANGES_REQUESTED** [review]
some language touch ups

**Comment by Termineitor244** [comment]
> some language touch ups

Thank you!

**Comment by I-am-Erk** [comment]
Sounds like we need an exception in the spaces checker for standard honourifics then. What a pain.

**Comment by a-chancey** [comment]
> > Kinda out of scope/off topic comment: It would be cool if those holyday themed things could spawn according to the start of the game:
> > ```
> > * winter start: plenty of santa and Xmass stuff around
> > 
> > * spring start: Xmass stuff are in boxes in basements
> > ```
> 
> Good ideas! I would have liked to implement the boxes already, but a quick search in the item groups folder didn't find the item group that boxes in basements use, maybe I should search in the mapgen folder.

https://github.com/CleverRaven/Cataclysm-DDA/blob/aac44034f5c24f9c9670e34f151ab207b2bf3a67/data/json/mapgen_palettes/basement.json#L200 That’s the line that starts the item sets for the palletized basements. 

https://github.com/CleverRaven/Cataclysm-DDA/tree/master/data/json/mapgen/basement That’s where the rest of the basements have things set

**Comment by Termineitor244** [comment]
> That’s the line that starts the item sets for the palletized basements.
> https://github.com/CleverRaven/Cataclysm-DDA/tree/master/data/json/mapgen/basement That’s where the rest of the basements have things set

Thanks! I will see to add these to the basements as well as some other (yet to be added) costumes in my next PR!

**Pull request merged**